halo2-base/src/virtual_region/mod.rs

```diff
@@ -0,0 +1,15 @@
+//! Trait describing the shared properties for a struct that is in charge of managing a virtual region of a circuit
+//! _and_ assigning that virtual region to a "raw" Halo2 region in the "physical" circuit.
+//!
+//! Currently a raw region refers to a subset of columns of the circuit, and spans all rows (so it is a vertical region),
+//! but this is not a requirement of the trait.
```

Comment:
I wasn't sure, because sometimes the same columns might have different gates overlapping etc. So it can't own column creation. Maybe gate creation is ok, but then for example you need to know how many columns you want (you can store it as a vector to get around this).

In general I wanted it to be easy to create the virtual managers without having to think too much about circuit shape. I also don't know if there's a good way in traits to impose these things.

For LookupManager my intention was LookupAnyManager is very bare bones, you can wrap it in other structs that contain the columns and lookup argument on a per-use basis.